File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/ActivityHandler.java
package closer.vlllage.com.closer.handler.helpers;

import android.app.Activity;

import closer.vlllage.com.closer.pool.PoolMember;

public class ActivityHandler extends PoolMember {

    private Activity activity;

    public boolean isPresent() {
        return this.activity != null && !this.activity.isDestroyed();
    }

    public Activity getActivity() {
        if (this.activity == null) {
            throw new IllegalStateException("Activity was not set!");
        }

        return activity;
    }

    public void setActivity(Activity activity) {
        if (this.activity != null) {
            throw new IllegalStateException("Cannot set Activity twice! Use another pool.");
        }

        this.activity = activity;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/handler/helpers/ActivityHandler.kt
package closer.vlllage.com.closer.handler.helpers

import android.app.Activity

import com.queatz.on.On

class ActivityHandler constructor(private val on: On) {

    var activity: Activity? = null
    set(value) {
        if (this.activity != null) {
            throw IllegalStateException("Cannot set Activity twice! Use another on.")
        }
        field = value
    }

    val isPresent: Boolean
        get() = this.activity != null && !this.activity!!.isDestroyed
}
